yt_dlp/extractor/megatvcom.py

```diff
@@ -0,0 +1,177 @@
+            for k, v in extract_attributes(player_el).items()
+            if k not in ('id',)
+        }
+
+
+class MegaTVComIE(MegaTVComBaseIE):
+    IE_NAME = 'megatvcom'
+    IE_DESC = 'megatv.com videos'
+    _VALID_URL = r'https?://(?:www\.)?megatv\.com/(?:\d{4}/\d{2}/\d{2}|[^/]+/(?P<id>\d+))/(?P<slug>[^/]+)'
+
+    _TESTS = [{
+        'url': 'https://www.megatv.com/2021/10/23/egkainia-gia-ti-nea-skini-omega-tou-dimotikou-theatrou-peiraia/',
+        'md5': '6546a1a37fff0dd51c9dce5f490b7d7d',
+        'info_dict': {
+            'id': '520979',
+            'ext': 'mp4',
+            'title': 'md5:70eef71a9cd2c1ecff7ee428354dded2',
+            'description': 'md5:0209fa8d318128569c0d256a5c404db1',
+            'timestamp': 1634975747,
+            'upload_date': '20211023',
+            'display_id': 'egkainia-gia-ti-nea-skini-omega-tou-dimotikou-theatrou-peiraia',
+            'thumbnail': 'https://www.megatv.com/wp-content/uploads/2021/10/ΠΕΙΡΑΙΑΣ-1024x450.jpg',
+        },
+    }, {
+        'url': 'https://www.megatv.com/tvshows/527800/epeisodio-65-12/',
+        'md5': 'cba2085d45c1abeb8e7e9b7e1d6c0072',
+        'info_dict': {
+            'id': '527800',
+            'ext': 'mp4',
+            'title': 'md5:fc322cb51f682eecfe2f54cd5ab3a157',
+            'description': 'md5:b2b7ed3690a78f2a0156eb790fdc00df',
+            'timestamp': 1636048859,
+            'upload_date': '20211104',
+            'display_id': 'epeisodio-65-12',
+            'thumbnail': 'https://www.megatv.com/wp-content/uploads/2021/11/16-1-1.jpg',
+        },
+    }]
+
+    def _real_extract(self, url):
+        video_id, display_id = self._match_valid_url(url).group('id', 'slug')
+        _is_article = video_id is None
+        webpage = self._download_webpage(url, video_id or display_id)
+        if _is_article:
+            video_id = self._search_regex(
+                r'<article[^>]*\sid=["\']Article_(\d+)["\']', webpage, 'article id')
+        player_attrs = self._extract_player_attrs(webpage)
+        title = player_attrs.get('label') or self._og_search_title(webpage)
+        description = get_element_by_class(
+            'article-wrapper' if _is_article else 'story_content',
+            webpage)
+        description = clean_html(re.sub(r'<script[^>]*>[^<]+</script>', '', description))
+        if not description:
+            description = self._og_search_description(webpage)
+        thumbnail = player_attrs.get('image') or self._og_search_thumbnail(webpage)
+        timestamp = unified_timestamp(self._html_search_meta(
+            'article:published_time', webpage))
+        source = player_attrs.get('source')
+        if not source:
+            raise ExtractorError('No source found', video_id=video_id)
+        if determine_ext(source) == 'm3u8':
+            formats, subs = self._extract_m3u8_formats_and_subtitles(source, video_id, 'mp4')
+        else:
+            formats, subs = [source], {}
```

Comment:
did u mean this?
```suggestion
            formats, subs = [{'url': source}], {}
```

Reply:
Oops, right. I should fix this elsewhere too.